### PR TITLE
Quieten output on success

### PR DESCRIPTION
**Pull request by vlsi**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
+distributionSha256Sum=7faa7198769f872826c8ef4f1450f839ec27f0b4d5d1e51bade63667cbccd205
```
[the remark this answers is not in the document]
Comment (autonomousapps) [reply gradle/wrapper/gradle-wrapper.properties:3]: Also, I suppose it's worth mentioning that this is orthogonal to the purpose of the PR and should probably be removed and done separately.

File: src/main/kotlin/com/autonomousapps/tasks/BuildHealthTask.kt [hunk 2]
```diff
@@ -61,14 +61,18 @@ abstract class BuildHealthTask : DefaultTask() {
...
       val consoleText = advicePrinter.consoleText()
-      buildHealthText
-        .appendReproducibleNewLine(projectHeaderText(projectAdvice.projectPath))
-        .appendReproducibleNewLine(consoleText)
-
-      // Only print to console if we're not configured to fail
-      if (!shouldFail) {
-        logger.quiet(projectHeaderText(projectAdvice.projectPath))
-        logger.quiet(consoleText)
+      if (consoleText.isNotEmpty()) {
```
Comment (autonomousapps) [inline src/main/kotlin/com/autonomousapps/tasks/BuildHealthTask.kt:64]: Note to self: run locally, check output.

@vlsi you might want to fix up `ProjectHealthTask` as well.

**Review by autonomousapps — CHANGES_REQUESTED** [review]
Thank you for contributing this! I like the direction, just a few small comments.

**Comment by autonomousapps** [comment]
You should rebase on `main`. I fixed (?) an issue that was causing the gradle daemon to die. Probably running out of memory.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlsi) [reply gradle/wrapper/gradle-wrapper.properties:3]: bin is fine for source navigation for ages: https://github.com/gradle/gradle/blob/master/gradle/wrapper/gradle-wrapper.properties#L3

[the remark this answers is not in the document]
Comment (eugene-krivobokov) [reply gradle/wrapper/gradle-wrapper.properties:3]: > bin is fine for source navigation for ages

Does it work as expected? 
We faced with a bug, probably this one: https://youtrack.jetbrains.com/issue/IDEA-231667

[the remark this answers is not in the document]
Comment (vlsi) [reply gradle/wrapper/gradle-wrapper.properties:3]: I do not understand what do you mean by "as expected".
I use -bin distributions for a long time, and I have no issues drilling down to Gradle sources

**Comment by autonomousapps** [comment]
My primary point is I'd prefer to discuss that separately, under another
PR. 

On June 13, 2021, GitHub ***@***.***> wrote:
> @vlsi commented on this pull request.
>

> In gradle/wrapper/gradle-wrapper.properties
> <https://github.com/autonomousapps/dependency-analysis-android-gradle-
> plugin/pull/402#discussion_r650549427>:
>

> @@ -1,5 +1,6 @@ distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists -
distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-
all.zip
+distributionSha256Sum=7faa7198769f872826c8ef4f1450f839ec27f0b4d5d1e51bade63667cbccd205

> I do not understand what do you mean by "as expected".
>  I use -bin distributions for a long time, and I have no issues
> drilling down to Gradle sources
>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/autonomousapps/dependency-analysis-android-gradle-
plugin/pull/402#discussion_r650549427>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABJG5PJ5IGUF54KUM2WLJQTTSTNFZANCNFSM45KADJNQ>.

**Comment by vlsi** [comment]
@autonomousapps , no problem, I can split two commits in two PRs, however, GitHub does not support [stacked diffs](https://twitter.com/acdlite/status/1255640994239766528), so splitting every commit into individual PR is really really painful.

For instance, no way I would build with `gradle-all.zip`, so I removed `gradle-bin` commit from here, however, now I skip testing at my side.

Don't get me wrong: I know it is a good practice to keep only related changes into a single PR/commit, however, I believe, it would take you a couple of minutes maximum to commit and even rebase the commits. You could cherry-pick the commits and close the PR. There are plenty of options, and all of them are way better than spending time in discussions here.

**Comment by autonomousapps** [comment]
I know this is tangential, but
> For instance, no way I would build with gradle-all.zip

Why? What makes the all distribution so bad you don't even want it touching your computer?

To the main point, one reason I've been sitting on this is that I thought the CI job failed. In fact, I'm almost certain I told github it was ok to run tests from this, but I just had to do it again 🤷

**Comment by autonomousapps** [comment]
Thanks for working with me on this, and addressing this long-standing issue.

**Review by autonomousapps — APPROVED** [review]

**Comment by autonomousapps** [comment]
In principle I approve these changes, but I'd still like to verify behavior locally before merging. Soon (and thanks).

**Comment by vlsi** [comment]
Thanks, I did try to find a test case (or a way to create such), however I failed, so no tests in PR 🤷‍♂️

**Comment by autonomousapps** [comment]
I deliberately try to avoid testing against output because it was too
volatile in the beginning. So I just have to see it in action. Thanks
for checking. 

On June 14, 2021, GitHub ***@***.***> wrote:
> Thanks, I did try to find a test case (or a way to create such),
> however I failed, so no tests in PR 🤷‍♂️
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/autonomousapps/dependency-analysis-android-gradle-
plugin/pull/402#issuecomment-861201968>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABJG5PJYKAKOLZ4NPC44MB3TS3VAVANCNFSM45KADJNQ>.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (eugene-krivobokov) [reply gradle/wrapper/gradle-wrapper.properties]: Probably, it depends on the project.

**Review by autonomousapps — CHANGES_REQUESTED** [review]
I noticed an issue with the console output.
```
> Task :buildHealth                                                                                                                                                          
Advice for project :app                                                                                                                                                      
Unused dependencies which should be removed:                                                                                                                                 
  - implementation(project(":entities"))                                                                                                                                     
  - implementation("com.google.dagger:dagger-android-support:2.24")                                                                                                          
  - implementation("androidx.core:core-ktx:1.0.1")                                                                                                                           
  - implementation("androidx.appcompat:appcompat:1.1.0-rc01")                                                                                                                
                                                                                                                                                                             
  Transitively used dependencies that should be declared directly as indicated:
  - implementation("com.google.dagger:dagger:2.24")
  - implementation("javax.inject:javax.inject:1")
                                                                                      
  Dependencies which could be compile-only:                          
  - compileOnly("androidx.annotation:annotation:1.1.0") (was implementation
```
`"Unused dependencies which should be removed:"` should be indented like the other headings.

**Comment by vlsi** [comment]
```kotlin
        val msg = projectHeaderText(projectAdvice.projectPath) + "\n  " + // <-- added indent here as well
          consoleText.replace("\n", "\n  ")
```

**Comment by vlsi** [comment]
@autonomousapps , what do you think if `- ` prefix was removed so the output is

```
> Task :buildHealth                                                                                                                                                          
Advice for project :app                                                                                                                                                      
  Unused dependencies which should be removed:                                                                                                                                 
    implementation(project(":entities"))                                                                                                                                     
    implementation("com.google.dagger:dagger-android-support:2.24")                                                                                                          
    implementation("androidx.core:core-ktx:1.0.1")                                                                                                                           
    implementation("androidx.appcompat:appcompat:1.1.0-rc01")                                                                                                                
                                                                                                                                                                             
  Transitively used dependencies that should be declared directly as indicated:
    implementation("com.google.dagger:dagger:2.24")
    implementation("javax.inject:javax.inject:1")
                                                                                      
  Dependencies which could be compile-only:                          
    compileOnly("androidx.annotation:annotation:1.1.0") (was implementation
```

Frankly speaking, it is exactly as readable, however, it would be easier to copy-paste (especially `Transitively used dependencies that should be declared directly`)

**Comment by autonomousapps** [comment]
Do it, sounds good to me. 

On June 20, 2021, GitHub ***@***.***> wrote:
> @autonomousapps <https://github.com/autonomousapps> , what do you
> think if - prefix was removed so the output would be
>
> > Task :buildHealth Advice for project :app Unused dependencies which
> should be removed: implementation(project(":entities"))
> implementation("com.google.dagger:dagger-android-support:2.24")
> implementation("androidx.core:core-ktx:1.0.1")
> implementation("androidx.appcompat:appcompat:1.1.0-rc01") Transitively
> used dependencies that should be declared directly as indicated:
> implementation("com.google.dagger:dagger:2.24")
> implementation("javax.inject:javax.inject:1") Dependencies which could
> be compile-only: compileOnly("androidx.annotation:annotation:1.1.0")
> (was implementation
> Frankly speaking, it is exactly as readable, however, it would be
> easier to copy-paste (especially Transitively used dependencies that
> should be declared directly)
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/autonomousapps/dependency-analysis-android-gradle-
plugin/pull/402#issuecomment-864921910>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABJG5PMJFRSY2Y2TIZQOQATTT4HZVANCNFSM45KADJNQ>.

**Comment by autonomousapps** [comment]
@vlsi just curious if you still want to work on this. No rush of course.

**Comment by autonomousapps** [comment]
New PR for same issue: https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/471

**Comment by autonomousapps** [comment]
Closing in favor of the other PR I just merged.

**Pull request closed without merging**